doc/tor.md

```diff
@@ -211,3 +209,96 @@ for normal IPv4/IPv6 communication, use:
   Otherwise it is trivial to link them, which may reduce privacy. Onion
   services created automatically (as in section 2) always have only one port
   open.
+
+- Some states have the ability to monitor and block use of the Tor network.
+  Recent examples include China and India. If you are in an environment that
+  does not permit direct Tor connections or the use of Tor bridges, then
+  considering the trade-offs, it may not be safe to use the Tor network.
+
+- For maximal privacy, disabling listening may be preferable to running an
+  onion service.
+
+- Users can download packages with [torsocks](https://gitlab.torproject.org/legacy/trac/-/wikis/doc/torsocks) for better security and privacy.
+
+- Trade-offs involved in using `onlynet=onion`:
```

Comment:
Here is how I would re-word this section. I feel that this encapsulates what you want to get across:

#### Trade-offs of the `onlynet=onion` configuration option:

The privacy benefits of Tor are affected if a Tor-Bitcoin service node is reachable by IPv4/IPv6 nodes. Placing `onlynet=onion` in your bitcoin.conf file will ensure that your node only attempts to connect over the Tor network, thus maximizing the benefits of using Tor. The decision to use this option should consider the following trade-offs:

**Sybil Attacks**
While Sybil attacks are a threat on clearnet networks, it is of greater concern on the Tor network. A node that only maintains connections over Tor is vulnerable to a Sybil attack because of the minimal cost of creating a Tor address. This minimal cost means that an attacker can flood the network with thousands of apparent Tor-Bitcoin nodes controlled by the attacker.

One way to mitigate this is by leaving out the `onlynet=onion` configuration option, which will allow connections to IPv4/IPv6 nodes. Clearnet IPs have a greater creation cost, making it harder for an attacker to carry out a successful Sybil attack. 

**Network Partitioning**
If every Tor-Bitcoin node ran using the `onlynet=onion` configuration option, no Tor node would communicate with the clearnet network. This would result in the Bitcoin network becoming partitioned. For the health of the Bitcoin network as a whole, some nodes must communicate among both clearnet and Tor. You can help prevent this scenario by excluding the `onlynet=onion` configuration option.

Reply:
No issues in rephrasing as long as we share important things with users in the docs. What are your thoughts on this suggestion? Cc: @sipa @jonatack

Reply:
Now that I2P support is merged, adding another privacy network, this may need some updating.

Reply:
Isn't the sybil concern partially mitigated, or at least changed, by the fact that Tor onion connection are anonymous? The attacker won't be able to pick you and sybil specifically you, potentially without you even knowing. 
The attacker will instead have to sybil the whole network, which is a much more complex and easier to spot thing.

Is correct to say that in this context there is a trade-off between targeted and non-targeted sybilling? And should it be mentioned?

Reply:
@jonatack which part? I2P will have a separate file `i2p.md` as discussed in other PR so this can still be focused on Tor and related things?

Reply:
> @jonatack which part?

the onlynet discussion

Reply:
> @jonatack which part?

The onlynet discussion and the "when to use tor" discussion... tor is no longer the only privacy network bitcoind can use, so for example, running onlynet=tor along with onlynet=i2p may be worth discussing. There may be reasons to prefer running both over onlynet=tor alone.

Reply:
> The onlynet discussion

Agree. This can be changed and will need comments from people who are more experienced to compare Tor and i2p usage in Bitcoin Core. TBH I have not used i2p much. Also not sure what should be added here and what will be included in `i2p.md`.

> "when to use/not use tor" discussion

The line added in this PR focuses on Tor and Tor bridges so I don't think it should be changed (with or without i2p reasons to avoid Tor in some parts remains the same). Maybe we can add one more "Privacy Recommendation" later in other PR.

Quoting the exact line I am talking about (Privacy Recommendation 2): 

_Some states have the ability to monitor and block use of the Tor network. Recent examples include China and India. If you are in an environment that does not permit direct Tor connections or the use of Tor bridges, then considering the trade-offs, it may not be safe to use the Tor network._

> tor is no longer the only privacy network bitcoind can use, so for example, running onlynet=tor along with onlynet=i2p may be worth discussing.

Agree Tor isnt the only option now but this PR and file `tor.md` is for Tor. 

> There may be reasons to prefer running both over onlynet=tor alone.

Maybe

Reply:
> Is correct to say that in this context there is a trade-off between targeted and non-targeted sybilling? And should it be mentioned?

@PulpCattel Not sure

Reply:
Made changes and added `onlynet=i2p`, also mentioned i2p as other option available for privacy in Bitcoin Core apart from Tor.